### PR TITLE
Add empty state feedback and animation to scanning process - Updates #4

### PR DESCRIPTION
**Pull request by Le-Roi777**
- Updates #4

- Added a loading state in the App component and passed it to the table components as prop (loadingRows). In the table component the following have been added to accommodate feedback for the scanning process.
- Added text feedback and loading indicator to the table component when rows are loading.
- Added initial state placeholder with an icon and text when no scan has been initialized. (if we don't have a response/rows).
- Added no results found feedback with an icon and text when the scan results are empty. (if we don't have a response/rows).
- Updates #3
- I also updated the README with development instructions : - )

empty-state

loading-state

### CODE REVIEW

**Comment by ibreakthecloud** [comment]
@Le-Roi777 Good stuff! Thanks for the PR. I'll review it.

File: ui/src/App.tsx [hunk 1]
```diff
@@ -72,47 +73,59 @@ export function App() {
...
                                 Deepfence SecretScanner can find unprotected secrets in container images or file
                                 systems.
                             </Typography>
 
-                            <Typography variant="body1" color="text.secondary" sx={{mt: 0}}>
+                            <Typography variant="body1" color="text.secondary" sx={{ mt: 0 }}>
                                 Select the image from the dropdown and scan for secrets.
                             </Typography>
+
+                            <Links />
```
Comment (ibreakthecloud) [inline ui/src/App.tsx]: why adding `<Links />` again?

Comment (Le-Roi777) [reply ui/src/App.tsx]: Hi @ibreakthecloud that was an merge conflict?

Comment (Le-Roi777) [reply ui/src/App.tsx]: My bad conflict was not resolved correctly. Not sure how it got there in the first place...

Comment (ibreakthecloud) [reply ui/src/App.tsx]: No worries.

**Review by ibreakthecloud — APPROVED** [review]
lgtm, thanks!

**Pull request merged**